src/sms-verification/scripts/directives.js

```diff
@@ -0,0 +1,64 @@
+(function (angular) {
+    'use strict';
+
+    var module = angular.module('user_management.sms-verification');
+
+    module.directive('smsVerify', [
+        '$parse',
+        '$routeParams',
+        'smsVerificationFactory',
+        function ($parse, $routeParams, smsVerificationFactory) {
+            return {
+                restrict: 'A',
+                scope: true,
+                templateUrl: 'templates/user_management/sms-verification/verify_form.html',
+                link: function (scope, element, attrs) {
+
+                    // 2-way bind data attribute to scope.
+                    // scope: { data: '=' }
+                    if (angular.isDefined(attrs.data)) {
```

Comment:
Passing in a `data` object, rather than two separate properties is more flexible for later additions, and keeps the html template tidier, although using separate code and phone number fields makes it more clear what the directive is doing.

Also passing in separate values ensures they are basic strings, so you don't have unintentional scope sharing problems using properties of an object.

In short, I'm not sure which is better :)